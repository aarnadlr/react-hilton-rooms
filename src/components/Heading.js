import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Main = styled.div`
  display: flex;
  justify-content: center;
  margin: 0 0 32px 0;
`;

const Input = styled.input`
  position: relative;
  top: 3px;
  margin: 0 8px 0 0;
`;

const Heading = ({ hasCheck, isChecked, roomNum = 0, onCheckboxChange }) => {
  return (
    <Main>
      {hasCheck && (
        // CHECKBOX
        <Input
          name="roomCheck"
          type="checkbox"
          checked={isChecked}
          onChange={onCheckboxChange}
          data-testid='heading-input-tag'
        />
      )}
      <header data-testid='heading-header-tag'>Room {roomNum}</header>
    </Main>
  );
};

Heading.propTypes = {
  hasCheck: PropTypes.bool.isRequired,
  roomNum: PropTypes.number.isRequired,
  isChecked: PropTypes.bool,
  onCheckboxChange: PropTypes.func
};

export default Heading;
