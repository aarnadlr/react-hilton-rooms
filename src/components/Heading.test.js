import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Heading from './Heading.js';

afterEach(cleanup);

it('renders correctly', () => {
  const { asFragment } = render(
    <Heading roomNum={1} hasCheck={false} isChecked={false} />
  );
  expect(asFragment()).toMatchSnapshot();
});
