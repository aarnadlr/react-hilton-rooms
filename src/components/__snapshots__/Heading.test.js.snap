// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa yFXSK"
  >
    <header
      data-testid="heading-header-tag"
    >
      Room 1
    </header>
  </div>
</DocumentFragment>
`;
